File: MVCPattern/Models/UserModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using DataProviderService;

namespace MVCPattern.Models
{
    public class UserModel
    {
        public int UserID { get; set; }
        public string FullName { get; set; }
        public string Gender { get; set; }
        public int Age { get; set; }

        public UserModel()
        {

        }

        public UserModel(DataProviderService.MyUser aUser)
        {
            UserID = aUser.UserID;
            FullName = aUser.FullName;
            Gender = aUser.Gender;
            Age = aUser.Age;
        }

        public MyUser ToMyUser()
        {
            MyUser temp = new MyUser() { UserID = this.UserID, FullName = this.FullName, Gender = this.Gender, Age = this.Age };

            return temp;
        }
    }
}

File: MVVMPattern/Model/UserModel.vb
Imports DataProviderService

Public Class UserModel
    Public Property UserID As Integer
    Public Property FullName As String
    Public Property Gender As String
    Public Property Age As Integer

    Public Sub New()

    End Sub

    Public Sub New(ByVal aRawUser As MyUser)
        Me.UserID = aRawUser.UserID
        Me.FullName = aRawUser.FullName
        Me.Gender = aRawUser.Gender
        Me.Age = aRawUser.Age
    End Sub

    Public Function ToMyUser() As MyUser
        ToMyUser = New MyUser() With {.UserID = Me.UserID, .FullName = Me.FullName, .Age = Me.Age, .Gender = Me.Gender}
    End Function

End Class
